### DISCUSSION TITLE
zoom element in imodel by iTwin Snapshot

### DISCUSSION THREAD

**Discussion by paprophet** [iTwin/itwinjs-core#1373] [Q&A]
now i have a bim file change from revit by iTwin Snapshot,but can not zoom some element( for example element id is 0x4A24)
but the element in the building  (in revit file)
Can I send the document to you for a  try?

**Comment by pmconne** [comment]
Explain what you mean by "can not zoom some element". If code is involved, show it.
Yes, you can send the file.

**Reply by paprophet** [reply]
but not every element can not zoom, some element can zoom,some can not .
the code like the demo from 
https://www.itwinjs.org/sample-showcase/?group=Viewer+Features&sample=zoom-to-elements-sample&imodel=Bay+Town+Process+Plant
 


private _handleZoomToElementsButton = async () => {
    const viewChangeOpts: ViewChangeOptions = {};
    if (this.state.animateEnable)
      viewChangeOpts.animateFrustumChange = this.state.animateVal;
    if (this.state.marginEnable)
      viewChangeOpts.marginPercent = new MarginPercent(this.state.marginVal, this.state.marginVal, this.state.marginVal, this.state.marginVal);

    const zoomToOpts: ZoomToOptions = {};
    if (this.state.relativeViewEnable)
      zoomToOpts.placementRelativeId = this.state.relativeViewVal;
    if (this.state.standardViewEnable)
      zoomToOpts.standardViewId = this.state.standardViewVal;

    const vp = IModelApp.viewManager.selectedView!;  
    let ids:string[]=[];
    ids.push("0x4A24");
   await ZoomToElementsAPI.zoomToElements(ids, viewChangeOpts, zoomToOpts, vp, this.state.imodel!);
  }


 public static async zoomToElements(elementIds: string[], viewChangeOpts: ViewChangeOptions, zoomToOpts: ZoomToOptions, vp: Viewport, imodel: IModelConnection) {

    // Set the view to point at a volume containing the list of elements
    await vp.zoomToElements(elementIds, { ...viewChangeOpts, ...zoomToOpts });

    // Select the elements.  This is not necessary, but it makes them easier to see.
    imodel.selectionSet.replace(elementIds);
  }

**Reply by paprophet** [reply]
pre-zoom:
![pre-zoom](https://user-images.githubusercontent.com/18022720/117818265-03e5d100-b29b-11eb-8746-7eade9435314.jpg)


after-zoom
![after-zoom](https://user-images.githubusercontent.com/18022720/117818231-fcbec300-b29a-11eb-954d-1c8db8c861ce.png)

**Comment by pmconne** [comment]
The element you mentioned (0x4a24) has no geometry. As your screenshot and my testing shows, the view **does** zoom to its location. There's just nothing to display at that location because the element has no geometry. Were you expecting it to have geometry?

**Reply by paprophet** [reply]
i will send the revit file to you 
in revit.i can see the element (revit element id is 1488985)

**Reply by pmconne** [reply]
Is it supposed to look like this?
![image](https://user-images.githubusercontent.com/22944042/117830035-45e22900-b241-11eb-98ec-867ee03b854a.png)

**Reply by paprophet** [reply]
yes

**Reply by pmconne** [reply]
I can find no element that looks like that in the iModel. I'll forward to the Revit connector folks.

**Reply by paprophet** [reply]
thank you !

**Reply by pmconne** [reply]
The connector team identified this as a known issue and is working on a fix.

**Reply by jffmarker** [reply]
We have reached out to our Revit library vendor to see how we can better support rebar. Revit doesn't store rebar geometry in their files most of the time, so geometry for them generally does not come across.

**Reply by jffmarker** [reply]
Our Revit Connector team says that RVT files do not store geometry for rebar elements. We would have to find a way to extract the rebar parameters from the RVT file and generate our own geometry to match. Product Management is currently evaluating whether to support this feature. I'd recommend raising the idea on https://itwinplatform.ideas.aha.io/ideas/ so that we know how many users are affected by this to prioritize it.

**Reply by candcconsulting** [reply]
Not quite the same issue, but similar in that the zoomToElements and emphasizeElements are dependent on IModelApp and selectionSet is dependent on IModelConnection ... is there some easier way to emphasize the elements ... ?
```
    onMouseButton(_ev: BeButtonEvent): boolean {
        if (!_ev.isDown) return true;
        const emph = EmphasizeElements.getOrCreate(IModelApp.viewManager.selectedView!);        
        IModelApp.viewManager.selectedView!.zoomToElements(this._elementId, { animateFrustumChange: true, standardViewId: StandardViewId.RightIso});
        emph.emphasizeElements(this._elementId, IModelApp.viewManager.selectedView!, undefined, true)
//        IModelConnection.selectionSet.replace(this._elementId)
        return true;
    }
```

**Reply by pmconne** [reply]
@candcconsulting to avoid cluttering up this unrelated topic, would you mind starting a separate question and providing more details about what you expect to happen vs what you are observing? Thanks.
